File: euler001/euler001.c
#include <stdio.h>
#include <string.h>

long arithSum(long b, long n)
{
    long N = (n - 1) / b;
    return b * (N * (N + 1)) / 2;
}

int main()
{
    int n;
    scanf("%d", &n);

    for (int i = 0; i < n; i++)
    {
        long maxNum;
        scanf("%lu", &maxNum);
        long sum = arithSum(3, maxNum) + arithSum(5, maxNum) - arithSum(15, maxNum);
        printf("%lu\n", sum);
    }
}

File: euler001/euler001.kt
import java.io.*
import java.util.*

fun main() {
        val N: Int = readLine()!!.toInt()

        for (i in 0..(N - 1)) {
            var n: Long = readLine()!!.toLong()
            var sum: Long = ArithSum(3, n) + ArithSum(5, n) - ArithSum(15, n)
            println(sum)
        }
}

fun ArithSum(b: Long, n: Long): Long {
    val N: Long = (n - 1) / b
    return b * (N * (N + 1)) / 2
}
